crates/aptos-rosetta/src/block.rs

```diff
@@ -166,30 +172,73 @@ impl BlockCache {
     }
 
     /// Retrieve the block info for the index
-    ///
-    /// TODO: Improve parallelism and performance
     pub async fn get_block_info(&self, block_index: u64) -> ApiResult<BlockInfo> {
         // If we already have the block info, let's roll with it
-        let (closest_known_block, closest_block_info): (u64, BlockInfo) = {
+        let (start_version, range, traverse_right) = {
             let map = self.blocks.read().unwrap();
             if let Some(block_info) = map.get(&block_index) {
                 return Ok(*block_info);
             }
 
-            // There will always be an index less than the index, since it starts with 0 set
-            let (index, block_info) = map.iter().rev().find(|(i, _)| **i < block_index).unwrap();
-            (*index, *block_info)
+            // Find the closest, left or right, and build up blocks from there
+            let search_left = map.iter().rev().find_map(|(i, info)| {
+                if *i < block_index {
+                    Some((i, info, block_index - i))
+                } else {
+                    None
+                }
+            });
+
+            let search_right = map.iter().find_map(|(i, info)| {
+                if *i > block_index {
+                    Some((i, info, block_index - i))
+                } else {
+                    None
+                }
+            });
```

Comment:
feels like you could probably do this in a single statement
```
let (search_left, search_right) = ... {
  if ... {
    (None, Some)
  } else if {
    (Some, None)
  } else {
    None
  }
});
```

Reply:
Sure, I can do that

Reply:
I just did it this way for readability